extensions/blocks/simple-payments/simple-payments.php

```diff
@@ -52,6 +53,17 @@ function render_block( $attr, $content ) {
 	$simple_payments = Jetpack_Simple_Payments::getInstance();
 	$simple_payments->enqueue_frontend_assets();
 
+	// For AMP requests, make sure the purchase link redirects to the non-AMP post URL.
+	if ( Blocks::is_amp_request() ) {
+		$non_amp_link = sprintf(
+			'<a href="%1$s" target="_blank" rel="noamphtml noopener noreferrer">%2$s</a>',
+			esc_url( get_permalink( get_the_ID() ) ),
+			__( 'View the non-AMP version to purchase.', 'jetpack' )
+		);
+		$content      = preg_replace( '#<a class="jetpack-simple-payments-purchase(.*)</a>#i', $non_amp_link, $content );
+		return $content;
+	}
```

Comment:
This won't work on sites in Standard mode, as in that case there is no non-AMP version to link to.

In that case, it would probably be preferable to prevent AMP entirely from being available for the post if this block is present on the post. This could be done using the `amp_skip_post` filter. For example:

```php
add_filter(
	'amp_skip_post',
	function ( $skip, $post_id, WP_Post $post ) {
		if ( false !== strpos( $post->post_content, '<!-- wp:jetpack/simple-payments' ) ) {
			$skip = true;
		}
		return $skip;
	},
	10,
	3
);
```

Reply:
The list of blocks with this situation will continue to grow; is skipping completely the best approach? Maybe we could check just for AMP Standard mode somehow?

Also curious, if a post is skipped, does that mean that post displays in the standard theme, while other posts display in AMP format?

Reply:
> The list of blocks with this situation will continue to grow; is skipping completely the best approach? Maybe we could check just for AMP Standard mode somehow?

Checking for standard mode can be done via `amp_is_canonical()`, so like:

```php
add_filter(
	'amp_skip_post',
	function ( $skip, $post_id, WP_Post $post ) {
		if ( amp_is_canonical() && false !== strpos( $post->post_content, '<!-- wp:jetpack/simple-payments' ) ) {
			$skip = true;
		}
		return $skip;
	},
	10,
	3
);
```

Ideally the block would be made AMP-compatible so that no such skipping or degradation in AMP would be necessary. This is more of a stop gap.

> Also curious, if a post is skipped, does that mean that post displays in the standard theme, while other posts display in AMP format?

If a post is skipped then AMP won't be available for that post. So on a Standard mode site, that one post would be not AMP while the other posts would be in AMP. For a site in Transitional mode or Reader mode (which are “Paired AMP”) then the `link[rel=amphtml]` would not be added to the canonical page: no AMP version would be available.

Reply:
Thanks @westonruter! I updated the PR with your suggestion and we now skip posts from AMP if the standard mode is enabled.

> Ideally the block would be made AMP-compatible so that no such skipping or degradation in AMP would be necessary.

Is there any way we can achieve that with a 3rd party script? The block uses the [PayPal checkout API](https://developer.paypal.com/docs/checkout/), but I guess that's not AMP-friendly?

Reply:
> Is there any way we can achieve that with a 3rd party script? The block uses the [PayPal checkout API](https://developer.paypal.com/docs/checkout/), but I guess that's not AMP-friendly?

Yeah, apparently the lack of AMP-compatibility is a known issue: https://github.com/paypal/paypal-checkout-components/issues/440

I chimed in on that issue to say that `amp-script` is now an option for adding custom JS to a page. But the problem is that the SDK is 271KB+ but `amp-script` imposes a 150KB limit.

Nevertheless, an `amp-iframe` could be an option. I tried putting the PayPal Checkout SDK in an `amp-iframe` in this Glitch: https://amp-paypal-checkout-components.glitch.me/

Currently the SDK is throwing an error which I don't understand:

> Error: Expected window to be same domain

Then again, I've never used the PayPal SDK before.

But the approach of using an `amp-iframe` to render the contents of a block has been done before in Jetpack. In particular, the Map block is doing this currently:

https://github.com/Automattic/jetpack/blob/f93ff480383680c0ad342fbdb14267952586e409/extensions/blocks/map/map.php#L93-L99